Mathlib/Data/Nat/Init.lean

```diff
@@ -434,6 +434,37 @@ protected lemma div_left_inj (hda : d ∣ a) (hdb : d ∣ b) : a / d = b / d ↔
   refine ⟨fun h ↦ ?_, congrArg fun b ↦ b / d⟩
   rw [← Nat.mul_div_cancel' hda, ← Nat.mul_div_cancel' hdb, h]
 
+lemma div_sub_of_dvd (h1 : 0 < n) (h2 : n ∣ m) : m / n - k = (m - n * k) / n := by
```

Comment:
Do you know about `Nat.sub_mul_div`? It looks a better version of your lemma (because it doesn't require divisibility), except that instead it needs `n * k ≤ m`, which you should be able to get away with.

In short, can you do
```suggestion
/-- TODO: Replace `Nat.sub_mul_div` in core. -/
lemma sub_mul_div' (x n p : Nat) : (x - n * p) / n = x / n - p := by
```
? Can you then perform the analogous changes for the other lemmas you're adding below?

Reply:
Thank you for the hint.

What do you mean by getting away with `n * k  ≤ m`? to use it as a hypothesis instead of the positivity and divisibility ones?

Another question: by doing the analogous changes you mean in the names of the theorems?

Reply:
Sorry, I mean that you should need neither divisibility nor `n * k ≤ m`.

By the analogous changes, I mean
1. Renaming the theorems
2. Renaming the variables
3. Turning the iff around
4. Remove the divisibility conditions

Reply:
I just pushed a commit with some of the suggested changes. I couldn't find more hypothesis to drop (nor relax, although I am not 100% sure about the existence of a possible weakening).

There exist some other theorems that give equivalences under weaker assumptions in the same spirit, but they produce more complicated formulae; so I think it is worth having these more concrete formulae for the cases where the divisions are exact.

Reply:
To add some context: the motivation for this PR comes from #8102 , where I try to implement a tactic that simplifies expressions in the same way that one would do naively by hand.